k8s.gcr.io/images/k8s-staging-e2e-test-images/images.yaml

```diff
@@ -14,6 +14,7 @@
     "sha256:8bd76a6a7778e203b61e18e5ee85b1faf52e6a30eb07e445a165ab90fe9633e3": ["2.31"]
     "sha256:758db666ac7028534dba72e7e9bb1e57bb81b8196f976f7a5cc351ef8b3529e1": ["2.32"]
     "sha256:5b3a9f1c71c09c00649d8374224642ff7029ce91a721ec9132e6ed45fa73fd43": ["2.33"]
+    "sha256:d0989420b6f0d2b929fd9355f15c767f62d0e9a72cdf999d1eb16e6073782c71": ["2.35"]
```

Comment:
I don't know what happened but my guess is that the image was rebuilt and pushed to staging after the PR was submitted. The digests in the manifest and in the registry do not match. Compare the above digest from the PR and the actual sha of the image:

```
crane digest gcr.io/k8s-staging-e2e-test-images/agnhost:2.35
sha256:3c5f7db4be8501eca4c5d0ce7bb3a1d890ce5b6ccfbe4db6fef7d9705a6b4434
```
@SergeyKanzhelev I think PRing the correct/new digest should trigger a new promotion and should copy the image.

Reply:
FYI `skipping gcr.io/k8s-staging-e2e-test-images/agnhost@sha256:d0989420b6f0d2b929fd9355f15c767f62d0e9a72cdf999d1eb16e6073782c71 because it is _LOST_ (can't find it in src registry!)` just means that the manifest is telling the promoter to promote `d0989420b6f0d2b929fd9355f15c767f62d0e9a72cdf999d1eb16e6073782c71` but it can't because it doesn't exist in gcr.io/k8s-staging-e2e-test-images/agnhost.

From the first comment it appears that this `d0989420b6f0d2b929fd9355f15c767f62d0e9a72cdf999d1eb16e6073782c71` digest is coming from

```
Unable to find image 'mplatform/mquery:latest' locally
latest: Pulling from mplatform/mquery
f7727aa0d57a: Pull complete 
2db63f065e4b: Pull complete 
Digest: sha256:d0989420b6f0d2b929fd9355f15c767f62d0e9a72cdf999d1eb16e6073782c71
```

which appears to be the wrong image (`mplatform/mquery`).

Reply:
Ah right, that's where it came from. And the correct digest is there:

```
Status: Downloaded newer image for mplatform/mquery:latest
Image: gcr.io/k8s-staging-e2e-test-images/agnhost:2.35 (digest: sha256:3c5f7db4be8501eca4c5d0ce7bb3a1d890ce5b6ccfbe4db6fef7d9705a6b4434)
 
```

Reply:
really sorry for this, perhaps a copy/paste issue. Thank you for helping!

Reply:
It shows me this digest:

```
docker run --rm mplatform/mquery gcr.io/k8s-staging-e2e-test-images/agnhost:2.35
Image: gcr.io/k8s-staging-e2e-test-images/agnhost:2.35 (digest: sha256:fde7ba8937b36702c2f840001d37ac15cc0cc401d0adcf5cc909f963f36b6f44)
```

And I can pull the image using this digest:

```
docker pull gcr.io/k8s-staging-e2e-test-images/agnhost@sha256:fde7ba8937b36702c2f840001d37ac15cc0cc401d0adcf5cc909f963f36b6f44
gcr.io/k8s-staging-e2e-test-images/agnhost@sha256:fde7ba8937b36702c2f840001d37ac15cc0cc401d0adcf5cc909f963f36b6f44: Pulling from k8s-staging-e2e-test-images/agnhost
8572bc8fb8a3: Already exists 
01c873a4c1a7: Pull complete 
3d2bf5363f5a: Pull complete 
802c6b4878a0: Pull complete 
6b328ba9b53c: Pull complete 
3b8e257bbe8d: Pull complete 
e0de9b1f2991: Pull complete 
0771f23944e0: Pull complete 
95571514db8c: Pull complete 
Digest: sha256:fde7ba8937b36702c2f840001d37ac15cc0cc401d0adcf5cc909f963f36b6f44
Status: Downloaded newer image for gcr.io/k8s-staging-e2e-test-images/agnhost@sha256:fde7ba8937b36702c2f840001d37ac15cc0cc401d0adcf5cc909f963f36b6f44
gcr.io/k8s-staging-e2e-test-images/agnhost@sha256:fde7ba8937b36702c2f840001d37ac15cc0cc401d0adcf5cc909f963f36b6f44
```

I also see this digest in the log: https://storage.googleapis.com/kubernetes-jenkins/logs/post-kubernetes-push-e2e-agnhost-test-images/1504270166685388800/artifacts/build.log


```
Created manifest list gcr.io/k8s-staging-e2e-test-images/agnhost:2.35
/workspace/test/images
sha256:fde7ba8937b36702c2f840001d37ac15cc0cc401d0adcf5cc909f963f36b6f44
PUSH
DONE
```

Reply:
Sent: https://github.com/kubernetes/k8s.io/pull/3532